pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -35,20 +36,26 @@
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
-# Programmer's guide 8-92, defined outside of the class, since it is used by
-# `Instrument.control` without access to `self`.
-MAX_MEASUREMENT_TIME = 1000
+# Defined outside of the class, since it is used by `Instrument.control` without access to `self`.
+MIN_GATE_TIME = 2e-9  # Programmer's guide 8-92
+MAX_GATE_TIME = 1000  # Programmer's guide 8-92
+MIN_BUFFER_SIZE = 4  # Programmer's guide 8-39
+MAX_BUFFER_SIZE = 10000  # Programmer's guide 8-39
 
 
 class CNT91(Instrument):
     """Represents a Pendulum CNT-91 frequency counter."""
 
     CHANNELS = {"A": 1, "B": 2, "C": 3, "E": 4, "INTREF": 6}
-    MAX_BUFFER_SIZE = 32000  # User Manual 8-38
 
     def __init__(self, adapter, name="Pendulum CNT-91", **kwargs):
-        kwargs.setdefault('timeout', 120000)
-        kwargs.setdefault('read_termination', '\n')
+        # allow long-term measurements, add 30 s for data transfer
+        kwargs.setdefault("timeout", 24 * 60 * 60 * 1000 + 30)
+        kwargs.setdefault("read_termination", "\n")
+        adapter = VISAAdapter(adapter, **kwargs)
+        # could not pass this to VISAAdapter, raises ValueError: 'baud_rate' is not a valid
+        # attribute for type USBInstrument
+        adapter.connection.baud_rate = 256000
```

Comment:
You have to use connection (as it belongs only to serial, not to USB etc. connection) specific arguments:

~~~python
super().__init__(
    adapter,
    name,
    asrl={'baud_rate': 256000}),
    **kwargs)
~~~

Reply:
Alternatively, you could just use `baud_rate=256000` in the super.__init__ call.
You should not create an adapter instance in the instrument's init, that is the problem.

Reply:
Thanks, that was the piece of documentation I could not find.

Reply:
I'm glad I could help